### PR TITLE
Collapsable Side Panel

### PR DESCRIPTION
**Pull request by SMahil**
- Add Planner Layout in widgets
- Integrate Side Panel to List View
- Integrate Side Panel to Board View
- Add Icons to Side Panel Titles
- CSS tweaks to Side Panel

### CODE REVIEW

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-6 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by vikram-raj** [comment]
screenshot from 2017-09-22 18-44-21

Quick add is not visible.

**Comment by vikram-raj** [comment]
screenshot from 2017-09-22 18-47-31

According to Ux design, these icons should be in the center.

screenshot from 2017-09-22 18-50-33

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-7 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-8 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-9 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-11 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-12 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-13 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-15 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

File: src/app/widgets/planner-layout/planner-layout.component.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  constructor(
+  ) { }
+
+  ngOnInit(): void {
+  }
+
+  ngAfterViewInit() {
+  }
+
+  test() {
```
Comment (vikram-raj) [inline src/app/widgets/planner-layout/planner-layout.component.ts]: Are you using this function?

File: src/app/widgets/planner-layout/planner-layout.component.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  sidePanelState: string = 'out';
+
+  constructor(
+  ) { }
+
+  ngOnInit(): void {
+  }
+
+  ngAfterViewInit() {
+  }
```
Comment (vikram-raj) [inline src/app/widgets/planner-layout/planner-layout.component.ts]: No need to implement Init functions if you are not using it.

File: src/app/widgets/planner-layout/planner-layout.component.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  @Input() sidePanelContent: TemplateRef<any>;
+  @Input() sectionContent: TemplateRef<any>;
+
+  @Output() sidePanelStateChange = new EventEmitter<string>();
+
+  contentHide: Boolean = false;
+  sidePanelState: string = 'out';
+
+  constructor(
+  ) { }
```
Comment (vikram-raj) [inline src/app/widgets/planner-layout/planner-layout.component.ts]: This should be in one line `constructor(){}`.

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-21 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-22 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-23 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-24 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by vikram-raj** [comment]
screenshot from 2017-10-04 12-34-59

Work Item quick adds not cover the remaining width. and tooltip for the collapsible icon is not visible.

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-25 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-26 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2152-28 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2152-fabric8-planner.badger.fabric8.io

**Pull request merged**